apiserver/common/networkingcommon/subnets.go

```diff
@@ -258,24 +258,21 @@ func (cache *addSubnetsCache) cacheSubnets(ctx context.ProviderCallContext) erro
 	return nil
 }
 
-// validateSubnet ensures either subnetTag or providerId is valid (not both),
+// validateSubnet ensures either cidr or providerId is valid (not both),
 // then uses the cache to validate and lookup the provider SubnetInfo for the
 // subnet, if found.
-func (cache *addSubnetsCache) validateSubnet(ctx context.ProviderCallContext, subnetTag, providerId string) (*network.SubnetInfo, error) {
-	haveTag := subnetTag != ""
+func (cache *addSubnetsCache) validateSubnet(ctx context.ProviderCallContext, cidr, providerId string) (*network.SubnetInfo, error) {
```

Comment:
I don't think this logic is correct any longer. We will need at least CIDR, but also provider ID when working in an environment that has the same CIDR occurring in different networks.